### PR TITLE
fix: Prevent sending message if not approved

### PR DESCRIPTION
**Pull request by boidushya**
# Description

Prevent sending message if not approved

# Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Fixes/Resolves (Optional)

#135 

# Examples/Screenshots (Optional)

| Mode       | Example        |
| ------------ | ------------ |
| Dark | image |
| Light | image |

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/messages/MessageBox/index.tsx [hunk 1]
```diff
@@ -13,6 +14,14 @@ const MessageBox: React.FC<MessageBoxProps> = ({ topic, authorAccount }) => {
   const [messageText, setMessageText] = useState('')
   const { chatClientProxy } = useContext(W3iContext)
 
+  const isDisabled = useMemo(
+    () =>
+      (topic.includes('invite:')
+        ? (topic.split(':')[1] as ChatClientTypes.SentInvite['status'])
+        : 'approved') !== 'approved',
+    [topic]
```
Comment (devceline) [inline src/components/messages/MessageBox/index.tsx:22]: You can use the `w3iProxy` and retrieve `sentInvites` from the context and query them to find the actual status

File: src/components/general/Textarea/index.tsx [hunk 2]
```diff
@@ -29,7 +36,8 @@ const Textarea: React.FC<ITextareaProps> = ({ onChange, value, icon, placeholder
         value={value}
         className="Textarea__textarea"
         onChange={onChange}
-        placeholder={placeholder}
+        disabled={disabled}
+        placeholder={disabled ? `You cannot send a message yet!` : placeholder}
```
[the remark this answers is not in the document]
Comment (boidushya) [reply src/components/general/Textarea/index.tsx:40]: Sure! That sounds good

File: src/components/messages/MessageBox/index.tsx (hunk 1 above)
Comment (boidushya) [reply src/components/messages/MessageBox/index.tsx:22]: Yep, gotcha!

File: src/components/messages/MessageBox/index.tsx [hunk 3]
```diff
@@ -47,19 +58,48 @@ const MessageBox: React.FC<MessageBoxProps> = ({ topic, authorAccount }) => {
...
-        <SendIcon />
-      </button>
+      {isDisabled ? (
+        <p className="MessageBox__status">
+          {inviteStatus === 'pending' ? (
+            <>Waiting to accept your invite…</>
+          ) : (
+            <>
+              Your chat invite was declined{' '}
+              <button className="MessageBox__status--btn">
```
Comment (devceline) [inline src/components/messages/MessageBox/index.tsx:68]: My bad, I should've given you a heads up but we're removing the retry button. :pray:  Please remove it

Comment (boidushya) [reply src/components/messages/MessageBox/index.tsx:68]: Ah! No problem :)

**Review by devceline — APPROVED** [review]
LGTM

**Pull request merged**